mantidproject/mantid
# Automatic calibration creation in ISIS_Powder PEARL scripts

A routine for generating grouping files from the `isis_powder` library for PEARL has been implemented. In addition, this has been kitted out with the usual trimmings of documentation and a system test.

**To test:**
Setup for the ISIS_Powder is a bit fiddly - come and get me to help you set up right directory structure.
First, make sure this script runs and gives you a sensible looking workspace:
```
from isis_powder import Pearl

pearl = Pearl(config_file="path/to/config.yaml")
pearl.create_cal(run_number="98494",
                 tt_mode="tt70",
                 long_mode=False)
```
Try fiddling with parameters (such as trying out different `run_number` formats, and see if anything goes bang.

Fixes #20877 

**Release Notes** 

[Here](https://github.com/mantidproject/mantid/commit/12a7f89e92f7f6e1e316e3fe8c1f528d02f059e6)

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
